src/OpenSage.Game/Audio/AudioSystem.cs

```diff
@@ -161,6 +161,20 @@ public void PlayAudioEvent(string eventName)
             PlayAudioEvent(audioEvent);
         }
 
+        public void DisposeSource(AudioSource source)
+        {
+            if (source == null)
+            {
+                return;
+            }
+
+            if (source.IsPlaying())
+            {
+                source.Stop();
+            }
+            _sources.Remove(source);
```

Comment:
This could crash if the AudioSource was not created by the AudioSystem

Reply:
No, `Remove` would only return `False` if the source was not in the list.